config/routes.rb

```diff
@@ -3,4 +3,9 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  get '/users', to: 'users#index', as: :users
+  get '/users/:id', to: 'users#show', as: :user
+  get '/users/:user_id/posts', to: 'posts#index', as: :user_posts
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: :user_post
```

Comment:
- [ ] [optional] I would suggest replacing the routes with the below code. It will make the code more readable and follow best practices.

```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end